lib/matplotlib/axis.py

```diff
@@ -2107,9 +2120,11 @@ def set_ticks_position(self, position):
         if position == 'top':
             self.set_tick_params(which='both', top=True, labeltop=True,
                                  bottom=False, labelbottom=False)
+            self._text_offset_position = 'top'
         elif position == 'bottom':
             self.set_tick_params(which='both', top=False, labeltop=False,
                                  bottom=True, labelbottom=True)
+            self._text_offset_position = 'bottom'
```

Comment:
Should we also set this in the `default` and `both` code paths below?  I can see some surprising hysteresis in interactive usage.

Reply:
Yes - to be fair, the original PR (#5243) did that, sorry for not having noticed